core/src/test/java/io/cucumber/core/options/CucumberOptionsAnnotationParserTest.java

```diff
@@ -218,6 +220,44 @@ void create_with_glue() {
             contains(uri("classpath:/app/features/user/registration"), uri("classpath:/app/features/hooks")));
     }
 
+    @Test
+    void warn_when_features_as_filesystem_path_to_resources() {
+        // warn when 'src/{test,main}/resources' is used
+
+        LogRecordListener logRecordListener = new LogRecordListener();
+        LoggerFactory.addListener(logRecordListener);
+
+        FeaturePath.parse("src/test/resources/features");
+        FeaturePath.parse("src/main/resources/features/feature1");
+
+        LoggerFactory.removeListener(logRecordListener);
+
+        assertAll(
+            () -> assertTrue(LogRecordListener.anyRecordMessageMatch(logRecordListener,
+                ".*replace.*src/test/resources/features.*'classpath:features'.*")),
+            () -> assertTrue(LogRecordListener.anyRecordMessageMatch(logRecordListener,
+                ".*replace.*src/main/resources/features/feature1.*'classpath:features/feature1'.*")));
+    }
+
+    @Test
+    void warn_when_glue_as_filesystem_path() {
+        // warn when 'src/{test,main}/{java,kotlin,scala,groovy}' is used
+
+        LogRecordListener logRecordListener = new LogRecordListener();
+        LoggerFactory.addListener(logRecordListener);
+
+        GluePath.parse("src/main/java/com/example");
+        GluePath.parse("src/test/java/com/package/other_package");
+
+        LoggerFactory.removeListener(logRecordListener);
+
+        assertAll(
+            () -> assertTrue(LogRecordListener.anyRecordMessageMatch(logRecordListener,
+                ".*replace.*src/main/java/com/example.*'com.example'.*")),
+            () -> assertTrue(LogRecordListener.anyRecordMessageMatch(logRecordListener,
+                ".*replace.*src/test/java/com/package/other_package.*'com.package.other_package'.*")));
```

Comment:
When this assertion fails, the assertion failed exception will not contain any useful information. Consider using matchers instead. They usually generate better exception messages. For example:

```
var records = logRecordListener.getLogRecords().stream()
  .map(LogRecord::getMessage)
  .collect(toList());
  
asserThat(records, containsInAnyOrder(matchesPattern(".*some-pattern.*)));
```  

See:

 * http://hamcrest.org/JavaHamcrest/javadoc/2.2/org/hamcrest/collection/IsIterableContainingInAnyOrder.html#containsInAnyOrder-org.hamcrest.Matcher...-
 * http://hamcrest.org/JavaHamcrest/javadoc/2.2/org/hamcrest/Matchers.html#matchesPattern-java.util.regex.Pattern-

Reply:
And unfortunately we're still using `Hamcrest`. If you're familiar with `AssertJ` feel free to use that.